app/components/network_indicator/network_indicator.js

```diff
@@ -70,6 +69,7 @@ export default class NetworkIndicator extends PureComponent {
 
         const navBar = this.getNavBarHeight(props.isLandscape);
         this.top = new Animated.Value(navBar - HEIGHT);
+        this.opacity = 0;
```

Comment:
Does not need to be in the state, and the fact that top is changing the opacity value is available during render

Reply:
Doesn't it need to be in the state to trigger a re-render then?

Reply:
Re-render is not triggered by changing the opacity value

Reply:
But shouldn't that cause a re-render so that the bar will appear/disappear when the opacity changes?

Reply:
Not needed.

Let me explain, when the props change and the component need to be shown we set the opacity as 1 while the component re-renders by changing the top value, and then it does the opposite when the component hides.

The whole idea to change the opacity is that the component won’t be visible when changing the orientation of the device

Reply:
That still seems a bit weird to not include it in the state in that case since the extra render wouldn't hurt then, but it's probably fine regardless.